crml/cennzx-spot/src/types.rs

```diff
@@ -44,6 +56,38 @@ impl FeeRate {
 		N::sa(lhs.as_() * SCALE_FACTOR / rhs.0)
 	}
 
+	/// Divide a `u128` supported numeric by a FeeRate
+	pub fn safe_div<N: Into<u128>>(lhs: N, rhs: FeeRate) -> rstd::result::Result<u128, &'static str> {
+		let lhs = lhs.into();
+		let lhs = U256::from(lhs);
+		let scale_factor = U256::from(SCALE_FACTOR);
+		let rhs = U256::from(rhs.0);
+		let res: u128 = (lhs * scale_factor / rhs).try_into().map_err(|_| "Overflow error")?;
+
+		Ok(res)
+	}
+
+	//Self - lhs and N - rhs
+	pub fn safe_mul<N: Into<u128>>(lhs: FeeRate, rhs: N) -> u128 {
```

Comment:
doesn't here also need to return `Result`? since we agree the fee rate can be > 1.

Reply:
As discussed made this change